modules/502-delivery/crds/werfsource.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: werfsources.deckhouse.io
+  labels:
+    heritage: deckhouse
+    module: delivery
+spec:
+  group: deckhouse.io
+  scope: Cluster
+  names:
+    plural: werfsources
+    singular: werfsource
+    kind: WerfSource
+  preserveUnknownFields: false
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          description: |
+            Delivery source configuration for Argo CD and werf bundle. It is used to create a registry
+            configuration for Argo CD Image Updater and optionally to create a OCI Helm repository
+            in Argo CD to use with werf bundles.
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - imageRepo
+              properties:
+                imageRepo:
```

Comment:
Does it work with http registries or https without a valid certificate / with selfsigned certificate (common cases in air-gapped environments)?

Reply:
Argo CD repo has options of SSH, HTTPS, or Github Apps. 
<img width="877" alt="image" src="https://user-images.githubusercontent.com/5133540/204284715-d1bae16e-9ec6-4fa5-8ffa-a52e48d128f8.png">

In general, there are options for insecure connection and self-signed certs. I'd leave it for follow-up features. 

https://argo-cd.readthedocs.io/en/latest/user-guide/commands/argocd_repo_add/#options

```
      --enable-lfs                              enable git-lfs (Large File Support) on this repository
      --enable-oci                              enable helm-oci (Helm OCI-Based Repository)
      --github-app-enterprise-base-url string   base url to use when using GitHub Enterprise (e.g. https://ghe.example.com/api/v3
      --github-app-id int                       id of the GitHub Application
      --github-app-installation-id int          installation id of the GitHub Application
      --github-app-private-key-path string      private key of the GitHub Application
  -h, --help                                    help for add
      --insecure-ignore-host-key                disables SSH strict host key checking (deprecated, use --insecure-skip-server-verification instead)
      --insecure-skip-server-verification       disables server certificate and host key checks
      --name string                             name of the repository, mandatory for repositories of type helm
      --password string                         password to the repository
      --project string                          project of the repository
      --proxy string                            use proxy to access repository
      --ssh-private-key-path string             path to the private ssh key (e.g. ~/.ssh/id_rsa)
      --tls-client-cert-key-path string         path to the TLS client cert's key path (must be PEM format)
      --tls-client-cert-path string             path to the TLS client cert (must be PEM format)
      --type string                             type of the repository, "git" or "helm" (default "git")
      --upsert                                  Override an existing repository with the same name even if the spec differs
      --username string                         username to the repository
```